hadoop-crypto/src/main/java/com/palantir/hadoop/PathConvertingFileSystem.java

```diff
@@ -49,7 +49,7 @@ public PathConvertingFileSystem(FileSystem delegate,
 
     @Override
     public URI getUri() {
-        return delegate.getUri();
+        return from(new Path(delegate.getUri())).toUri();
```

Comment:
why is this required? (do you have test that fails if you used `delegate.getUri()`?)